File: Codeforces root folder2/Crafting.py
n = int(input())
for pos in range(n):
    size = input()
    arr = [int(e) for e in input().split(' ')]
    tar = [int(e) for e in input().split(' ')]

    totalReduce = 0
    for i in range(len(arr)):
        reduce = max(tar[i]- arr[i], 0)
        totalReduce += reduce
    # print(totalReduce)
    for i in range(len(arr)):
        reduce = max(tar[i] - arr[i], 0)
        curVal = max(tar[i], arr[i]) - (totalReduce - reduce)
        if curVal < tar[i]:
            print('NO')
            break
    else:
        print('YES')

File: Codeforces root folder 2/Crafting.cpp
#include<iostream>
#include<unordered_map>
#include<algorithm>
using namespace std;
int main() {
  int t;
  cin >> t;
  while (t--) {
    int size;
    cin >> size;
    int arr[size];
    int tar[size];
    for (int i = 0; i < size; i++) {
      cin >> arr[i];
    }
    for (int i = 0; i < size; i++) {
      cin >> tar[i];
    }
    long long totalReduce = 0;
    for (int i = 0; i < size; i++) {
      int reduce = max(tar[i] - arr[i], 0);
      totalReduce += reduce;
    }

    bool valid = true;

    for (int i = 0; i < size; i++) {
      int reduce = max(tar[i] - arr[i], 0);
      long long curVal = max(tar[i], arr[i]) - (totalReduce - reduce);
      if (curVal < tar[i]) {
        valid = false;    
        break;
      }
    }
    if (valid) {
      cout << "YES" << endl;
    }
    else {
      cout << "NO" << endl;
    }
  }
  
}